File: maple-engine-plugins/maple-spark/apps/maple-spark-data-calc/src/main/scala/org/xi/maple/datacalc/api/MapleTransform.scala
package org.xi.maple.datacalc.api

import org.apache.spark.sql.{Dataset, Row, SparkSession}
import org.xi.maple.datacalc.model.TransformConfig

trait MapleTransform[T <: TransformConfig] extends MaplePlugin[T] with Logging {
  def process(spark: SparkSession, ds: Dataset[Row]): Dataset[Row]
}

File: maple-engine-plugins/maple-flink/apps/maple-flink-data-calc/src/main/java/org/xi/maple/datacalc/flink/api/MapleTransform.java
package org.xi.maple.datacalc.flink.api;

import org.apache.flink.table.api.TableEnvironment;
import org.xi.maple.datacalc.flink.model.MaplePluginConfig;

public abstract class MapleTransform<T extends MaplePluginConfig> extends MaplePlugin<T> implements TableDefine {

    public MapleTransform(TableEnvironment tableEnv) {
        super(tableEnv);
    }
}
